### DISCUSSION TITLE
Errors in Installing Pyleecan

### DISCUSSION THREAD

**Discussion by TangiPilabz** [Eomys/pyleecan#708] [Q&A]
Hi,
I am very new to Pyleecan and I am encountering many errors while installing pyleecan on my windows system.  The steps I followed are from the readme.md page on Github.com/Eomys/pyleecan. 

Basically, I am not able to install all the dependencies in requirements-full.txt. 

As suggested in the documentation I used Python 3.8 version.  PySide2 is deprecated so unable to get through it.
Can anyone tell me the basic steps to install all the dependencies and build pyleecan from source successfully.
"pip install pyleecan" gives many errors as well.  I also tried "python setup.py install" nothing worked.  

Please help and let me know the steps to install pyleecan.

Thanks, and Regards
TangiPilabz

**Comment by BonneelP** [comment]
Hello,

Can you share the errors that you get from "pip install pyleecan" ? Did you try with requirements.txt instead of requirements-full.txt ?

Best regards,
Pierre

**Reply by TangiPilabz** [reply]
Hello,

Sorry, yesterday I finally figured out the issue and could install pyleecan successfully using python 3.8 and pip 24.2 by  running the command "pip install -r requirements.txt" Earlier I was using an older version of pip. 
But when I use requirements-full.txt I am getting the following errors:
_________________________________________________
-packages\gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean
      copying gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean\import.geo -> build\bdist.win-amd64\wheel\gmsh_sdk-4.6.0.post1.data\data\Lib\site-packages\gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean
ackages\gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean
      copying gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean\intersect_line_volume.geo -> build\bdist.win-amd64\wheel\gmsh_sdk-4.6.0.post1.data\data\Lib\site-packages\gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean
      C:\Users\srinivas-0020\AppData\Local\Temp\pip-build-env-40d7ncgd\overlay\Lib\site-packages\setuptools\dist.py:294: InformationOnly: Normalizing '4.6.0-1' to '4.6.0.post1'
        self.metadata.version = self._normalize_version(self.metadata.version)
      error: could not create 'build\bdist.win-amd64\wheel\gmsh_sdk-4.6.0.post1.data\data\Lib\site-packages\gmsh-4.6.0-Windows64-sdk\share\doc\gmsh\demos\boolean\intersect_line_volume.geo': No such file or directory
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for gmsh-sdk
  Building wheel for grapheme (pyproject.toml) ... done
  Created wheel for grapheme: filename=grapheme-0.6.0-py3-none-any.whl size=210116 sha256=4dbe73f8384096ca0a1126417b11b0695d8ce1c2ef08c3ac74bfde55cdb55f42       
  Stored in directory: c:\users\srinivas-0020\appdata\local\pip\cache\wheels\2d\08\6b\126ea9009f7482fd53a78d0db2ece5aca70af8f4a30445386b
Successfully built smoot pymoo grapheme
Failed to build gmsh-sdk
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (gmsh-sdk)
PS D:\MyPyleecan\pyleecan-master\pyleecan-master>

**Reply by BonneelP** [reply]
Seems that the issue is linked to gmsh-sdk. As long as you don't use a gmsh related feature it should not be an issue to use pyleecan, you can focus on the FEMM coupling.
